_drafts/postgres-operations-that-you-cant-explain.md

```diff
@@ -0,0 +1,282 @@
+---
+layout: post
+title: "Postgres operations that you can't EXPLAIN"
+---
+
+I am currently dabbling in postgres internals in my spare time, including
+trying to understand the query planner and executor. It's a deeply humbling
+experience, but occasionally I'm delighted to be able to answer simple
+questions I'm researching.
+
+One of these questions is how postgres is executing `ORDER BY` clauses inside
+of aggregate calls.
+
+Consider a simple table made of 3 integers, 1...3:
+
+```
+postgres=# CREATE TABLE foo AS
```

Comment:
TIL about `CREATE TABLE AS` :) But honestly, this multi-line `CREATE TABLE foo AS` tripped me up. I now know how it works, but the fact that you show a prompt for the `generate_series` line made me think that it's multiple commands send to the DB.

I'd say, either show what the prompt would really look like:

```
mrnugget@[local]:5432 mrnugget# CREATE TABLE foo AS
SELECT * FROM generate_series(1, 3) i;
SELECT 3
mrnugget@[local]:5432 mrnugget# SELECT * FROM foo;
 i
---
 1
 2
 3
(3 rows)

mrnugget@[local]:5432 mrnugget#
```

Or put it on a single line:

```
mrnugget@[local]:5432 mrnugget# CREATE TABLE foo AS (SELECT * FROM generate_series(1, 3) i);
SELECT 3
mrnugget@[local]:5432 mrnugget# SELECT * FROM foo;
 i
---
 1
 2
 3
(3 rows)
```